links/src/commonMain/kotlin/com/splendo/kaluga/links/manager/DefaultLinksManager.kt

```diff
@@ -0,0 +1,46 @@
+/*
+ Copyright 2020 Splendo Consulting B.V. The Netherlands
+
+    Licensed under the Apache License, Version 2.0 (the "License");
+    you may not use this file except in compliance with the License.
+    You may obtain a copy of the License at
+
+      http://www.apache.org/licenses/LICENSE-2.0
+
+    Unless required by applicable law or agreed to in writing, software
+    distributed under the License is distributed on an "AS IS" BASIS,
+    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+    See the License for the specific language governing permissions and
+    limitations under the License.
+
+ */
+
+package com.splendo.kaluga.links.manager
+
+import com.splendo.kaluga.links.utils.decodeFromList
+import kotlinx.serialization.KSerializer
+
+class DefaultLinksManager(
+    private val linksHandler: LinksHandler
+) : LinksManager {
+
+    override fun <T> handleIncomingLink(url: String, serializer: KSerializer<T>): T? {
+        val list = linksHandler.extractQueryAsList(url)
+        if (list.isEmpty()) {
+            return null
+        }
+
+        return decodeFromList(list, serializer)
+    }
+
+    override fun validateLink(url: String): String? {
+        return if (linksHandler.isValid(url)) {
+            url
+        } else {
+            null
+        }
+    }
+}
+
+expect class LinksManagerBuilder
```

Comment:
Not sure I understand, which problem platform-specific `LinksManagerBuilder` solve.

Reply:
Makes sense, it is there from old implementation. 750e374 d4ef7ed